numpy/lib/npyio.py

```diff
@@ -1107,22 +1107,19 @@ def read_data(chunk_size):
         for i in range(skiprows):
             next(fh)
 
-        # Read until we find a line with some values, and use
-        # it to estimate the number of columns, N.
-        first_vals = None
-        try:
-            while not first_vals:
-                first_line = next(fh)
-                first_vals = split_line(first_line)
-        except StopIteration:
-            # End of lines reached
+        # Read until we find a line with some values, and use it to determine
+        # the need for decoding and estimate the number of columns.
+        for first_line in fh:
+            ncols = len(usecols or split_line(first_line))
```

Comment:
This logic is not the same when the file is empty; previously it set `ncols to `len(usecols)`, but now it sets it to zero.

Reply:
fair enough, now fixed.